datamade/bga-payroll
# Integrate with django-salsa-auth

## Description

This PR:

- Removes the existing login functionality.
- Integrates with [`django-salsa-auth`](https://github.com/hancush/django-salsa-auth).
- Containerizes the core application for local development and updates `after_install` to use `docker-compose.deployment.yml`.

## Testing instructions

- Create your own settings file: `cp bga_database/local_settings.py.example bga_database/local_settings.py`. Fill in valid test email and Salsa credentials.
- Run the application, as described in the README. Run a few searches – you won't have any data, the point is to trigger the authentication modal. Confirm the modal pops up after 5 unauthenticated searches.
- Click sign up and enter your information. Be sure to use an email you have access to! Confirm the form submits and you receive a message to check your email.
- Go to your email and click the verification link. Confirm that you are redirected to the homepage with a welcome back message.
- Run more searches. Confirm that you aren't re-prompted for your information.
- In a private browser window (to remove cookies), repeat the first steps, and use your email to sign in. Confirm that you can, and that you're redirected to the homepage with a welcome back message.
- Go back to your email and grab the activation link. Fudge a few characters and try to visit it. Confirm that you are redirected to the homepage with an invalid link message.
